sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-text.service.ts

```diff
@@ -0,0 +1,124 @@
+import { Injectable } from "@nestjs/common";
+import { EducationProgramService, InstitutionLocationService } from "..";
+import { plainToClass } from "class-transformer";
+import { validateSync } from "class-validator";
+import { flattenErrorMessages } from "../../utilities/class-validation";
+import { removeUTF8BOM } from "../../utilities";
+import { LoggerService, InjectLogger } from "@sims/utilities/logger";
+import {
+  ApplicationBulkWithdrawalData,
+  ApplicationBulkWithdrawalFooter,
+  ApplicationBulkWithdrawalHeader,
+  ApplicationWithdrawalTextModel,
+  ApplicationWithdrawalTextValidationResult,
+  RecordType,
+} from "./application-bulk-withdrawal-text.models";
+import { APPLICATION_WITHDRAWAL_INVALID_TEXT_FILE_ERROR } from "../../constants";
+import { CustomNamedError } from "@sims/utilities";
+
+/**
+ * Handles the application withdrawal bulk insert preparation.
+ */
+@Injectable()
+export class ApplicationWithdrawalImportTextService {
+  constructor(
+    private readonly institutionLocationService: InstitutionLocationService,
+    private readonly educationProgramService: EducationProgramService,
+  ) {}
+
+  /**
+   * Reads a text content and generate a text object model for each line.
+   * @param textContent text content to generate the models.
+   * @returns text object models.
+   */
+  readText(textContent: string): ApplicationWithdrawalTextModel[] {
+    const applicationWithdrawalModels: ApplicationWithdrawalTextModel[] = [];
+    // Remove BOM(Byte order mark), if present.
+    textContent = removeUTF8BOM(textContent);
+    const fileLines = textContent.split("\n");
```

Comment:
@guru-aot , In some languages like Python, we don't use split for file parsing, because there will be issues with large file, I am not aware of typescript parsing with large content, Did u test with large files?

Reply:
Ive not tested with large file, but with some 100 data set I have tested. BUt for your question, I guess we are using split for quite a lot of integrations like msfaa and sftp. SO I feel it should not be a problem.